src/components/PopupWithForm.js

```diff
@@ -19,8 +20,11 @@ export class PopupWithForm extends Popup {
   setEventListeners() {
     this._popupForm.addEventListener("submit", (event) => {
       event.preventDefault();
+      const submitText = this._submitButton.textContent;
+      this._submitButton.textContent = "Saving...";
       this._data = this._getInputValues();
       this._handleSubmit(event, this._data);
+      this._submitButton.textContent = submitText;
```

Comment:
You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.